Dependency problems with Angular / Jest

### Current Behavior

Updating an Angular NX workspace to 21.3.1 or 21.3.2 OR creating a new workspace causes NPM dependency conflicts with Jest.

### Expected Behavior

I expect a newly created workspace to install without problems.

### GitHub Repo

_No response_

### Steps to Reproduce

Create a new Angular workspace with those parameters:

```
> npx create-nx-workspace@21.3.2 nx-angular-app --preset=angular-monorepo

√ Application name · nx-angular-app
√ Which bundler would you like to use? · esbuild
√ Default stylesheet format · scss
√ Do you want to enable Server-Side Rendering (SSR) and Static Site Generation (SSG/Prerendering)? · No
√ Which unit test runner would you like to use? · jest
√ Test runner to use for end to end (E2E) tests · cypress
√ Which CI provider would you like to use? · skip
√ Would you like remote caching to make your build faster? · skip
```

Run npm install and you get those errors:

```
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: jest-preset-angular@14.6.1
npm error Found: jest@30.0.5
npm error node_modules/jest
npm error   dev jest@"^30.0.2" from the root project
npm error   peer jest@"^29.0.0 || ^30.0.0" from ts-jest@29.4.0
npm error   node_modules/ts-jest
npm error     dev ts-jest@"^29.4.0" from the root project
npm error     ts-jest@"^29.3.0" from jest-preset-angular@14.6.1
npm error     node_modules/jest-preset-angular
npm error       dev jest-preset-angular@"~14.6.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer jest@"^29.0.0" from jest-preset-angular@14.6.1
npm error node_modules/jest-preset-angular
npm error   dev jest-preset-angular@"~14.6.0" from the root project
npm error
npm error Conflicting peer dependency: jest@29.7.0
npm error node_modules/jest
npm error   peer jest@"^29.0.0" from jest-preset-angular@14.6.1
npm error   node_modules/jest-preset-angular
npm error     dev jest-preset-angular@"~14.6.0" from the root project
...
```


### Nx Report

```shell
Node           : 22.15.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.9.2

nx (global)            : 21.3.2
nx                     : 21.3.2
@nx/js                 : 21.3.2
@nx/jest               : 21.3.2
@nx/eslint             : 21.3.2
@nx/workspace          : 21.3.2
@nx/angular            : 21.3.2
@nx/cypress            : 21.3.2
@nx/devkit             : 21.3.2
@nx/eslint-plugin      : 21.3.2
@nx/module-federation  : 21.3.2
@nx/rspack             : 21.3.2
@nx/web                : 21.3.2
@nx/webpack            : 21.3.2
typescript             : 5.8.3
---------------------------------------
Registered Plugins:
@nx/cypress/plugin
@nx/eslint/plugin
---------------------------------------
Community plugins:
angular-eslint : 20.1.1
---------------------------------------
Cache Usage: 0.00 B / 190.61 GB
```

### Failure Logs

```shell

```

### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [x] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

Even after installing `jest-preset-angular v15` you get

```
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @angular-devkit/build-angular@20.1.1
npm error Found: jest@30.0.5
npm error node_modules/jest
npm error   dev jest@"30.0.5" from the root project
npm error   peer jest@"^30.0.0" from jest-preset-angular@15.0.0
npm error   node_modules/jest-preset-angular
npm error     dev jest-preset-angular@"15.0.0" from the root project
npm error   1 more (ts-jest)
npm error
npm error Could not resolve dependency:
npm error peerOptional jest@"^29.5.0" from @angular-devkit/build-angular@20.1.1
npm error node_modules/@angular-devkit/build-angular
npm error   dev @angular-devkit/build-angular@"20.1.1" from the root project
npm error   peerOptional @angular-devkit/build-angular@">= 18.0.0 < 21.0.0" from @nx/angular@21.3.3
npm error   node_modules/@nx/angular
npm error     @nx/angular@"21.3.3" from the root project
npm error
npm error Conflicting peer dependency: jest@29.7.0
npm error node_modules/jest
npm error   peerOptional jest@"^29.5.0" from @angular-devkit/build-angular@20.1.1
npm error   node_modules/@angular-devkit/build-angular
npm error     dev @angular-devkit/build-angular@"20.1.1" from the root project
npm error     peerOptional @angular-devkit/build-angular@">= 18.0.0 < 21.0.0" from @nx/angular@21.3.3
npm error     node_modules/@nx/angular
npm error       @nx/angular@"21.3.3" from the root project
```

--- a/docs/generated/manifests/new-nx-api.json
+++ b/docs/generated/manifests/new-nx-api.json
@@ -399,6 +399,16 @@
       }
     },
     "migrations": {
+      "/technologies/angular/api/migrations/21.3.4-jest-package-updates": {
+        "description": "",
+        "file": "generated/packages/angular/migrations/21.3.4-jest-package-updates.json",
+        "hidden": false,
+        "name": "21.3.4-jest-package-updates",
+        "version": "21.3.4-beta.0",
+        "originalFilePath": "/packages/angular",
+        "path": "/technologies/angular/api/migrations/21.3.4-jest-package-updates",
+        "type": "migration"
+      },
       "/technologies/angular/api/migrations/update-angular-cli-version-20-1-0": {
         "description": "Update the @angular/cli package version to ~20.1.0.",
         "file": "generated/packages/angular/migrations/update-angular-cli-version-20-1-0.json",

--- a/docs/generated/packages-metadata.json
+++ b/docs/generated/packages-metadata.json
@@ -440,6 +440,16 @@
     ],
     "migrations": [
       {
+        "description": "",
+        "file": "generated/packages/angular/migrations/21.3.4-jest-package-updates.json",
+        "hidden": false,
+        "name": "21.3.4-jest-package-updates",
+        "version": "21.3.4-beta.0",
+        "originalFilePath": "/packages/angular",
+        "path": "angular/migrations/21.3.4-jest-package-updates",
+        "type": "migration"
+      },
+      {
         "description": "Update the @angular/cli package version to ~20.1.0.",
         "file": "generated/packages/angular/migrations/update-angular-cli-version-20-1-0.json",
         "hidden": false,

--- a/docs/generated/packages/angular/migrations/21.3.4-jest-package-updates.json
+++ b/docs/generated/packages/angular/migrations/21.3.4-jest-package-updates.json
@@ -1,0 +1,23 @@
+{
+  "name": "21.3.4-jest-package-updates",
+  "version": "21.3.4-beta.0",
+  "requires": {
+    "@angular/compiler-cli": ">=18.0.0 <21.0.0",
+    "@angular/core": ">=18.0.0 <21.0.0",
+    "@angular/platform-browser-dynamic": ">=18.0.0 <21.0.0",
+    "jest": "^30.0.0"
+  },
+  "packages": {
+    "jest-preset-angular": {
+      "version": "~15.0.0",
+      "alwaysAddToPackageJson": false
+    }
+  },
+  "aliases": [],
+  "description": "",
+  "hidden": false,
+  "implementation": "",
+  "path": "/packages/angular",
+  "schema": null,
+  "type": "migration"
+}

--- a/packages/angular/migrations.json
+++ b/packages/angular/migrations.json
@@ -1883,6 +1883,21 @@
           "alwaysAddToPackageJson": false
         }
       }
+    },
+    "21.3.4-jest": {
+      "version": "21.3.4-beta.0",
+      "requires": {
+        "@angular/compiler-cli": ">=18.0.0 <21.0.0",
+        "@angular/core": ">=18.0.0 <21.0.0",
+        "@angular/platform-browser-dynamic": ">=18.0.0 <21.0.0",
+        "jest": "^30.0.0"
+      },
+      "packages": {
+        "jest-preset-angular": {
+          "version": "~15.0.0",
+          "alwaysAddToPackageJson": false
+        }
+      }
     }
   }
 }

--- a/packages/angular/src/utils/versions.ts
+++ b/packages/angular/src/utils/versions.ts
@@ -28,7 +28,7 @@ export const autoprefixerVersion = '^10.4.0';
 export const tsNodeVersion = '10.9.1';
 export const lessVersion = '^4.3.0';
 
-export const jestPresetAngularVersion = '~14.6.0';
+export const jestPresetAngularVersion = '~15.0.0';
 export const typesNodeVersion = '18.16.9';
 export const jasmineMarblesVersion = '^0.9.2';
 

